### ISSUE TITLE
New Rule: Use `!=` over `<>`

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#2398]
So strictly `!=` isn't ANSI standard whereas `<>` is, however, this is an extremely common style guide suggestion.

- Mazar's: https://github.com/mattm/sql-style-guide#use--over-

Simply because `!=` reads like "not equal" which is closer to how we'd say it out loud.

```sql
-- Good
select 
    count(*) as paying_users_count
from users
where 
    plan_name != 'free'
```

- Brooklyn Data co.: https://github.com/brooklyn-data/co/blob/main/sql_style_guide.md#use--instead-of-

`!=` reads like "not equal" which is closer to how we'd say it out loud.

- GitLab: https://github.com/brooklyn-data/co/blob/main/sql_style_guide.md#use--instead-of-

Prefer `!=` to `<>`. This is because `!=` is more common in other programming languages and reads like "not equal" which is how we're more likely to speak.

Given it's not ANSI we should check which dialects don't support this, but as far as I'm aware `!=` is pretty common so we can just exclude any dialects that don't support it 😄
